edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/IMessageProducer.cs

```diff
@@ -0,0 +1,8 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter
+{
+    public interface IMessageProducer
+    {
+        void SetConnector(IMqttBridgeConnector connector);
```

Comment:
Do we need this? Why can't the connector be passed in as part of initialization?

Reply:
That is because a circular dependency between the instances. The BridgeConnector has a reference to all IMessageProducers, and every message producer has a reference to the BridgeConnector.  It is a bit like the BindDeviceProxy() of the DeviceMessageHandler class

Reply:
How is the producer used in the MqttBridgeConnector? I don't see any usage other than to set the connector.

Reply:
They just come with in the same packet with the discovery object (discovery contains all components). I review that pattern if this can be eliminated

Reply:
I made an attempt to remove the SetConnector call, but it is not that simple. Certain classes implement both IMessageConsumer and IMessageProducer. The connector needs to know all the message consumers (because it has a loop asking all of them to process incoming messages). On the other hand MessageProducers need to know about the connector, so they can send out messages. If a class implements both interfaces, then it has a circular dependency with the connector